### DISCUSSION TITLE
Delete trigger by word

### DISCUSSION THREAD

**Discussion by eugenesvk** [espanso/espanso#1654] [Feature requests and Ideas]
As far as I understand, currently triggered text is deleted by character (but at least you can insert text wholesale via clipboard), and this slower than ideal and also pollutes undo stack a bit more than necessary
What if the triggers could be analyzed on load and then instead of backspacing ":manycharacterstotrigger" you could <kbd>⌥</kbd><kbd>⌫</kbd> (or whatever user-configurabe per-system keybind would delete back by word) in just two keypresses (another one to delete `:`) 

(and since espanso tracks which app triggers an expansion you could even have the perfect solution in advanced apps like vim that allow you to jump to arbitrary text: send keys that would trigger a command to select untill `:` and delete)

Another bright idea given potential app-dependent imprecisions: you could select by word, then cut (saving current clipboard not to lose it), compare how much you've cut vs. the originally inserted text and repeat untill everything is deleted (or even insert back is you've deleted a little extra) (and restoring clipboard in the end). This verification method could also help have the word jumping be less precise

**Comment by smeech** [comment]
Just to further the discussion - would this handle multi-word triggers, e.g. `- trigger: :two words`, and text fields/editors that have different methods of back-deletion?

**Reply by eugenesvk** [reply]
Sure, multiple words are no different just like multiple chars are currently requiring multiple backspaces.

Though re. differences in apps it might be a good start to make the interaction on a per app basic user configurable not to burden Espanso with having to track all the apps (some apps also have different word separators, so even the same word selection/word deletion keybind would not delete the same amount of text).

Also check out the update in the end with an idea on how Espanso could do some verification of how much it's deleted vs the trigger text

**Reply by smeech** [reply]
I can see the current character-by-character method is slow (is that a problem? and is the issue with the “undo stack” significant?). However, it fairly straightforwardly handles almost all characters (excepting CTRL-characters), including multiple spaces, in a way that generally works across different programs. That being said, Espanso does, relatively frequently, inaccurately delete trigger phrases, leaving the first character, or inserting an extraneous space.
